# CSMODEL Machine Project

This Jupyter notebook presents a comprehensive analysis of the [Family Income and Expenditure Survey 2012 Vol. 1](https://psada.psa.gov.ph/auth/register) dataset from the Philippine Statistics Authority (PSA).

The goal of the project is to explore how **socioeconomic and demographic factors** may be associated with the **expenditure patterns** of Filipino households in 2012.

**Research Question**
1. [How are socioeconomic and demographic factors associated with the expenditure patterns of Filipino households in 2012?](#how-are-socioeconomic-and-demographic-factors-associated-with-the-expenditure-patterns-of-filipino-households-in-2012)

**Exploratory Data Analysis (EDA) Questions**
1. [How does the proportion of spending across **major expenditure categories** (e.g., food, vices) vary based on **socioeconomic and demographic factors** (e.g., region, income bracket)?](#1-how-does-the-proportion-of-spending-across-major-expenditure-categories-vary-based-on-socioeconomic-factors-eg-region-income-bracket-family-size-sex-of-the-household-head-and-education-level-of-the-household-head)
2. [Which regions spend the highest proportion of their **food expenditure** on various **food categories** (e.g., vegetables, meats)?](#2-which-regions-spend-the-highest-proportion-of-their-food-expenditure-on-various-food-categories-eg-vegetables-meats)
3. [Is there a correlation between **total household income** and the proportion of food expenditure spent on **food consumed outside the home**?](#3-is-there-a-correlation-between-total-household-income-and-the-proportion-of-food-expenditure-spent-on-food-consumed-outside-the-home)
4. [How does **housing expenditure** differ between **urban** and **rural** households?](#4-how-does-housing-expenditure-differ-between-urban-and-rural-households)
5. [Is there a correlation between **total household income** and **education-related expenditures**?](#5-is-there-a-correlation-between-total-household-income-and-education-related-expenditures)

## Authors

The following students of De La Salle University - Manila, Philippines collaborated on this project:

<table>
  <thead>
    <tr>
      <th>Profile</th>
      <th>Author</th>
      <th>Contributions</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td align="center">
        <img src="https://github.com/qu1r0ra.png" width="50" height="50" style="border-radius: 50%;" />
      </td>
      <td>
        <strong>Christian Joseph Bunyi</strong>
        <br />
        <a href="https://github.com/qu1r0ra">@qu1r0ra</a>
      </td>
      <td>
        <ul>
          <li>Created and maintained the GitHub repository and the Jupyter notebook</li>
          <li>Constructed the research question</li>
          <li>Constructed EDA questions 1, 2, and 3</li>
          <li>Performed data cleaning and preprocessing (Section II)</li>
          <li>Performed EDA on EDA questions 1 and 2 (Section III)</li>
          <li>Wrote introductory and skeletal Markdown (Introduction, Authors, etc.)</li>
          <li>Wrote Markdown for Sections II and III</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td align="center">
        <img src="https://github.com/kelliekaw.png" width="50" height="50" style="border-radius: 50%;" />
      </td>
      <td>
        <strong>Kellie Kaw</strong>
        <br />
        <a href="https://github.com/kelliekaw">@kelliekaw</a>
      </td>
      <td>
        <ul>
          <li>[indicate contributions here]</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td align="center">
        <img src="https://github.com/JohnathanTantanan.png" width="50" height="50" style="border-radius: 50%;" />
      </td>
      <td>
        <strong>Lance Xavier Lim</strong>
        <br />
        <a href="https://github.com/JohnathanTantanan">@JohnathanTantanan</a>
      </td>
      <td>
        <ul>
          <li>[indicate contributions here]</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td align="center">
        <img src="https://github.com/jstnsy.png" width="50" height="50" style="border-radius: 50%;" />
      </td>
      <td>
        <strong>Justin John Abraham Sy</strong>
        <br />
        <a href="https://github.com/jstnsy">@jstnsy</a>
      </td>
      <td>
        <ul>
          <li>[indicate contributions here]</li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>

## I. Dataset

```
Note from CJ: Employ a first person, narrative tone--as if we're guiding readers along the notebook.
```

### Description (Justin)

>[write stuff]

### Data collection method (Justin)

>[write stuff]

### Structure (Kellie)

>[write stuff]

## II. Data Cleaning and Preprocessing (Lance and CJ)

Now that we have a good understanding of the dataset and how it was collected, we can proceed with cleaning and preprocessing it.

Cleaning the data is crucial to avoid errors or unexpected results later on, which may result from data that is inconsistent, incorrect, missing, etc.

First, let us import all the Python libraries and modules which we will be using throughout the notebook. Brief descriptions of the purpose of each library/module are indicated as comments.

In [39]:
import pandas as pd         # brief description of purpose

Next, we load the dataset from a `.csv` file. The **pandas** library is ideal for this as it optimized for handling tabular data like that from the survey.

In [40]:
df = pd.read_csv('data/FIES_PUF_2012_Vol_1.csv')

It's good practice to view high level information of a dataset when looking at it the first time. `df.info()` allows us to do so.

In [41]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40171 entries, 0 to 40170
Columns: 119 entries, W_REGN to REGPC
dtypes: float64(5), int64(92), object(22)
memory usage: 36.5+ MB


From `df.info()`, we can learn that the dataset indeed contains $40171$ entries or rows or **observations** and $119$ columns or **attributes**.

We can also learn that:
- $5$ attributes are of datatype float64
- $92$ attributes are of datatype int64
- $22$ attributes are of datatype object

It also helps looking at some sample observations to see how data is structured and encoded. `df.head()` allows us to do so.

In [42]:
df.head()

W_REGN      W_OID  W_SHSN  W_HCN  URB   RSTR     PSU  BWEIGHT     RFACT  \
0      14  101001000       2     25    2  21100  415052   138.25  200.6576   
1      14  101001000       3     43    2  21100  415052   138.25  200.6576   
2      14  101001000       4     62    2  21100  415052   138.25  200.6576   
3      14  101001000       5     79    2  21100  415052   138.25  200.6576   
4      14  101001000      10    165    2  21100  415052   138.25  200.6576   

   FSIZE  ...  PC_QTY  OVEN_QTY  MOTOR_BANCA_QTY  MOTORCYCLE_QTY   POP_ADJ  \
0    3.0  ...      01        01                                   0.946172   
1   12.5  ...                01                               01  0.946172   
2    2.0  ...                01                                   0.946172   
3    4.0  ...                01                                   0.946172   
4    5.0  ...                                                 01  0.946172   

       PCINC  NATPC  NATDC  REGDC  REGPC  
0  108417.00      9      8      8      9  
1   30631.60      5      9      9      4  
2   86992.50      9      6      6      8  
3   43325.75      6      6      6      6  
4   37481.80      6      6      6      5  

[5 rows x 119 columns]

From `df.head()`, we can learn that the dataset's attribute names follow some coding scheme which does not clearly convey their meaning. This is not a problem, as we can refer to the **metadata dictionary** provided along with the dataset. Short descriptions of each attribute can also be seen in [Section I](#structure-kellie).

Next, we will drop unnecessary attributes. This step is dependent on our [research question](#research-question) and [EDA questions](#eda-questions) as they determine which variables are needed and which ones are irrelevant given the scope our EDA.

>**Note:**
>From this point, the authors used the [Data Wrangler](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.datawrangler) extension in Visual Studio Code to make data cleaning and viewing easier.

[dropping columns explanation]

In [43]:
# Drop columns: 'W_OID', 'W_SHSN' and 81 other columns
df = df.drop(columns=['W_OID', 'W_SHSN', 'W_HCN', 'RSTR', 'PSU', 'RFACT', 'BWEIGHT', 'FSIZE', 'AGRI_SAL', 'NONAGRI_SAL', 'WAGES', 'NETSHARE', 'CASH_ABROAD', 'CASH_DOMESTIC', 'RENTALS_REC', 'INTEREST', 'PENSION', 'DIVIDENDS', 'OTHER_SOURCE', 'NET_RECEIPT', 'REGFT', 'NET_LPR', 'NET_CFG', 'NET_FISH', 'NET_FOR', 'NET_RET', 'NET_MFG', 'NET_COM', 'NET_TRANS', 'NET_MIN', 'NET_CONS', 'NET_NEC', 'EAINC', 'LOSSES', 'T_ACTRENT', 'T_RENTVAL', 'T_IMPUTED_RENT', 'T_BIMPUTED_RENT', 'T_OTHREC', 'T_TOREC', 'FOOD_ACCOM_SRVC', 'MS', 'JOB', 'OCCUP', 'KB', 'CW', 'HHTYPE', 'MEMBERS', 'AGELESS5', 'AGE5_17', 'EMPLOYED_PAY', 'EMPLOYED_PROF', 'SPOUSE_EMP', 'BLDG_TYPE', 'ROOF', 'WALLS', 'TENURE', 'HSE_ALTERTN', 'TOILET', 'ELECTRIC', 'WATER', 'DISTANCE', 'RADIO_QTY', 'TV_QTY', 'CD_QTY', 'STEREO_QTY', 'REF_QTY', 'WASH_QTY', 'AIRCON_QTY', 'CAR_QTY', 'LANDLINE_QTY', 'CELLPHONE_QTY', 'PC_QTY', 'OVEN_QTY', 'MOTOR_BANCA_QTY', 'MOTORCYCLE_QTY', 'POP_ADJ', 'PCINC', 'NATPC', 'NATDC', 'REGDC', 'REGPC']);

In [ ]:
# for debugging!
df.head()

W_REGN  URB   TOINC  T_BREAD  T_MEAT  T_FISH  T_MILK  T_OIL  T_FRUIT  \
0      14    2  325251    30263   29374    5204    3533   2136     2129   
1      14    2  382895    97693    5820   28836   14484   3016     1230   
2      14    2  173985    14699   13457    7811    1222   1788     2610   
3      14    2  173303    27846    7787    5106    4112   1713     1530   
4      14    2  187409    29573   11704    8765    4056   1138     1385   

   T_VEG  ...  T_EDUCATION  T_MISCELLANEOUS  T_OTHER_EXPENDITURE  \
0   6517  ...        41200            18636                  260   
1   7885  ...         1180            10260                    0   
2   5560  ...            0            10224                 1615   
3   5435  ...          430            10020                   15   
4   4250  ...        10600             5274                    0   

   T_OTHER_DISBURSEMENT  T_NFOOD  T_TOTEX  T_TOTDIS  SEX  AGE  HGC  
0                     0   198916   313269    313269    2   75  280  
1                     0    61108   246625    246625    1   48  250  
2                  5000    86321   140966    145966    1   55  552  
3                     0    84601   156661    156661    1   44  830  
4                     0   108002   179855    179855    1   42  350  

[5 rows x 37 columns]

In [ ]:


# Change column type to string for columns: 'W_REGN', 'URB' and 1 other column
df = df.astype({'W_REGN': 'string', 'URB': 'string', 'SEX': 'string'});

# Replace all instances of "41" with "4A" in column: 'W_REGN'
df.loc[df['W_REGN'].str.lower() == "41".lower(), 'W_REGN'] = "4A"

# Replace all instances of "42" with "4B" in column: 'W_REGN'
df.loc[df['W_REGN'].str.lower() == "42".lower(), 'W_REGN'] = "4B"

# Replace all instances of "1" with "Urban" in column: 'URB'
df.loc[df['URB'].str.lower() == "1".lower(), 'URB'] = "Urban"

# Replace all instances of "2" with "Rural" in column: 'URB'
df.loc[df['URB'].str.lower() == "2".lower(), 'URB'] = "Rural"

# Replace all instances of "1" with "Male" in column: 'SEX'
df.loc[df['SEX'].str.lower() == "1".lower(), 'SEX'] = "Male"

# Replace all instances of "2" with "Female" in column: 'SEX'
df.loc[df['SEX'].str.lower() == "2".lower(), 'SEX'] = "Female"

# The codes are taken from the metadata dictionary.
def convert_hgc_code_to_string(code):
    exact_matches = {
        0: 'No Grade Completed',
        10: 'Preschool',
        280: 'Elementary Graduate',
        350: 'High School Graduate',
        900: 'Post Baccalaureate'
    }

    if code in exact_matches:
        return exact_matches[code]
    elif 210 <= code <= 260:
        return 'Elementary Undergraduate'
    elif 310 <= code <= 330:
        return 'High School Undergraduate'
    elif 410 <= code <= 420:
        return 'Post Secondary'
    elif 501 <= code <= 589:
        return 'Post Secondary / Technical Vocational Graduate'
    elif 810 <= code <= 840:
        return 'College Undergraduate'
    elif 601 <= code <= 689:
        return 'College Graduate'
    else:
        return 'N/A'

df['HGC'] = df['HGC'].apply(convert_hgc_code_to_string);

Now that we have cleaned the dataset, we will proceed to preprocessing by applying necessary transformations (e.g., *encoding*, *normalization*, *standardization*). This prepares it for [**exploratory data analysis (EDA)**](#iii-exploratory-data-analysis-eda).

In [ ]:
# TODO: Create new attributes for the dataframe based on data we will need for EDA
def engineer_attributes():
    pass

## III. Exploratory Data Analysis (EDA)

### Research Question

#### How are socioeconomic and demographic factors associated with the expenditure patterns of Filipino households in 2012?

### EDA Questions

>[write stuff]

#### 1. How does the proportion of spending across major expenditure categories vary based on socioeconomic factors (e.g., region, income bracket, family size, sex of the household head, and education level of the household head)?

>[write stuff]

#### 2. Which regions spend the highest proportion of their food expenditure on various food categories (e.g., vegetables, meats)?

>[write stuff]

#### 3. Is there a correlation between total household income and the proportion of food expenditure spent on food consumed outside the home?

>[write stuff]

#### 4. How does housing expenditure differ between urban and rural households?

>[write stuff]

#### 5. Is there a correlation between total household income and education-related expenditures?

>[write stuff]

<br>

## IV. Data Mining

To be continued for phase 2.

<br>

## V. Statistical Inference

To be continued for phase 2.

<br>

## VI. Insights and Conclusions

To be continued for phase 2.

<br>

## Sources and Citations

During the preparation of this work, the authors used [NAME TOOL/SERVICE]
for the following purposes:

- [purposes]

After using this tool, the authors reviewed and edited the content as needed and takes
full responsibility for the content of the publication.